src/test/kotlin/cacophony/controlflow/generation/StructTest.kt

```diff
@@ -145,32 +132,298 @@ class StructTest {
+         */
+        val fDef =
+            unitFunctionDefinition(
+                "f",
+                listOf(),
+                block(
+                    variableDeclaration("x", simpleStruct()),
+                    variableUse("x") assign struct("b" to lit(false), "a" to lit(5)),
+                    empty(),
+                ),
+            )
+
+        // when
+        val actualCFG = generateSimplifiedCFG(fDef)
+
+        // then
+        val virA = Register.VirtualRegister()
+        val virB = Register.VirtualRegister()
+
+        val expectedCFG =
+            singleWrappedFragmentCFG(fDef) {
+                "bodyEntry" does jump("assign b") { writeRegister(virA, integer(7)) }
+                "assign b" does jump("reassign a") { writeRegister(virB, integer(1)) }
+                "reassign a" does jump("reassign b") { writeRegister(virA, integer(5)) }
+                "reassign b" does jump("assign res") { writeRegister(virB, integer(0)) }
+                "assign res" does jump("bodyExit") { writeRegister(getResultRegister(), integer(42)) }
+            }
+
+        assertEquivalent(actualCFG, expectedCFG)
+    }
+
+    @Test
+    fun `reassign in nested`() {
+        // given
+
+        /*
+         * let f = [] -> Int => (let x = {a = 5, b = {a = 7, b = true} }; x.b = {a = 5, b = false}; x.b.a);
+         */
+        val fDef =
+            intFunctionDefinition(
+                "f",
+                listOf(),
+                block(
+                    variableDeclaration("x", nestedStruct()),
+                    lvalueFieldRef(variableUse("x"), "b") assign simpleStruct(5, false),
+                    rvalueFieldRef(rvalueFieldRef(variableUse("x"), "b"), "a"),
+                ),
+            )
+
+        // when
+        val actualCFG = generateSimplifiedCFG(fDef)
+
+        // then
+        val virA = Register.VirtualRegister()
+        val virB = Register.VirtualRegister()
+        val virC = Register.VirtualRegister()
+
+        val expectedCFG =
+            singleWrappedFragmentCFG(fDef) {
+                "bodyEntry" does jump("assign b.a") { writeRegister(virA, integer(5)) }
+                "assign b.a" does jump("assign b.b") { writeRegister(virB, integer(7)) }
+                "assign b.b" does jump("reassign b.a") { writeRegister(virC, integer(1)) }
+                "reassign b.a" does jump("reassign b.b") { writeRegister(virB, integer(5)) }
+                "reassign b.b" does jump("assign res") { writeRegister(virC, integer(0)) }
+                "assign res" does jump("bodyExit") { writeRegister(getResultRegister(), registerUse(virB)) }
             }
 
         assertEquivalent(actualCFG, expectedCFG)
     }
+
+    // TODO: fix this test - accessing subfield makes whole structure land on stack
+    @Test
+    fun `access simple subfield in nested function`() {
+        // given
+
+        /*
+         * let f = [] -> Int => (
```

Comment:
I copied this code and try to compile it, got an exception at instruction covering - is this related to the issue you mentioned or is something else yet to be done?

Reply:
I have no idea, can you provide more info (DM me, or here)?

Reply:
There is a new issue (#253) addressing that.